.github/workflows/unity-build.yml

```diff
@@ -0,0 +1,49 @@
+name: Unity Build and Test
+
+on:
+  # Build every commit to 'main'
+  push:
+    branches:
+      - main
+
+  # Build every PR
+  pull_request:
+
+jobs:
+  test:
+    name: Unity Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+
+      - name: Activate Problem Matchers
+        run:
+          echo "::add-matcher::$GITHUB_WORKSPACE/.github/problemMatchers/unity-compile.json"
+
+      - name: Retrieve Cached Library
+        uses: actions/cache@v3
+        with:
+          path: Library
+          key: Library-${{ hashFiles('Assets/**', 'Packages/**', 'ProjectSettings/**') }}
+          restore-keys: |
+            Library-
+
+      - name: Run Unity Tests
+        id: run-unity-tests
+        uses: game-ci/unity-test-runner@v4
+        env:
+          UNITY_LICENSE: ${{ secrets.UNITY_LICENSE }}
+        with:
+          projectPath: src/RGUnityBots
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
So right now this will only run the tests in the SDK, correct? And in the future how should we run tests that are in the games? Also I am guessing we will have a README in this project that mentions things like this?